src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -231,6 +235,47 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
         });
     }
 
+    /**
+     * Parse tasks and corresponding test cases from problem statement for this exercise
+     */
+    extractTasksFromProblemStatement(): void {
+        this.programmingExerciseService.extractTasksFromProblemStatement(this.programmingExercise.id!).subscribe({
+            next: (res) => {
+                const numberTests = res.map((task) => task.tests.length).reduce((numberTests1, numberTests2) => numberTests1 + numberTests2, 0);
+                this.alertService.addAlert({
+                    type: 'success',
+                    message: 'artemisApp.programmingExercise.extractTasksFromProblemStatementSuccess',
+                    translationParams: { numberTasks: res.length, numberTestCases: numberTests, detailedResult: this.buildTaskCreationMessage(res) },
+                    // long timeout in order to test properly
+                    timeout: 10000000,
+                });
+            },
+            error: (error) => this.dialogErrorSource.next(error.message),
+        });
+    }
+
+    private buildTaskCreationMessage(tasks: Task[]): string {
+        let message = '';
+        message += tasks.map((task) => '"' + task.taskName + '": ' + task.tests).join('\n');
+        return message;
```

Comment:
```suggestion
        return tasks.map((task) => '"' + task.taskName + '": ' + task.tests).join('\n');
```
Shouldn't this be possible? If it returns undefined for some reason you could also write
`return tasks.map((task) => '"' + task.taskName + '": ' + task.tests).join('\n') || '';`